proto/regen/group/v1alpha1/types.proto

```diff
@@ -0,0 +1,262 @@
+    google.protobuf.Any decision_policy = 6 [(cosmos_proto.accepts_interface) = "DecisionPolicy"];
+}
+
+// Proposal defines the core field members of a group proposal.
+message Proposal {
+    option (gogoproto.goproto_getters) = false;
+
+    // group_account is the group account address.
+    bytes group_account = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    
+    // comment is the proposal's comment.
+    string comment = 2;
+
+    // proposers are the account addresses of the proposers.
+    repeated bytes proposers = 3 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    
+    // submitted_at is a timestamp specifying when a proposal was submitted.
+    google.protobuf.Timestamp submitted_at = 4 [(gogoproto.nullable) = false];
+    
+    // group_version tracks the version of the group that this proposal corresponds to.
+    // When group membership is changed, existing proposals for prior group versions will become invalid.
+    uint64 group_version = 5;
+
+    // group_account_version tracks the version of the group account that this proposal corresponds to.
+    // When a decision policy is changed, an existing proposals for prior policy versions will become invalid.
+    uint64 group_account_version = 6;
+
+    // Status defines proposal statuses.
+    enum Status {
+        option (gogoproto.goproto_enum_prefix) = false;
+        
+        // An empty value is invalid and not allowed.
+        STATUS_UNSPECIFIED = 0 [(gogoproto.enumvalue_customname) = "ProposalStatusInvalid"];
+        
+        // Initial status of a proposal when persisted.
+        STATUS_SUBMITTED = 1 [(gogoproto.enumvalue_customname) = "ProposalStatusSubmitted"];
+        
+        // Final status of a proposal when the final tally was executed.
+        STATUS_CLOSED = 2 [(gogoproto.enumvalue_customname) = "ProposalStatusClosed"];
+        
+        // Final status of a proposal when the group was modified before the final tally.
+        STATUS_ABORTED = 3 [(gogoproto.enumvalue_customname) = "ProposalStatusAborted"];
+
+        // TODO: do we want to support a withdrawn operation?
+        // A proposal can be deleted before the voting start time by the owner. When this happens the final status
+        // is Withdrawn.
+        //        STATUS_WITHDRAWN = 4 [(gogoproto.enumvalue_customname) = "Withdrawn"];
+    }
+
+    // Status represents the high level position in the life cycle of the proposal. Initial value is Submitted.
+    Status status = 7;
+
+    // Result defines types of proposal results.
+    enum Result {
+        option (gogoproto.goproto_enum_prefix) = false;
+
+        // An empty value is invalid and not allowed
+        RESULT_UNSPECIFIED = 0 [(gogoproto.enumvalue_customname) = "ProposalResultInvalid"];
+        
+        // Until a final tally has happened the status is undefined
+        RESULT_UNDEFINED = 1 [(gogoproto.enumvalue_customname) = "ProposalResultUndefined"];
```

Comment:
not really, `RESULT_UNSPECIFIED` 0 is just a result value that would make the proposal invalid while `RESULT_UNDEFINED` is valid and means that the proposal result is not defined because a final tally hasn't happened yet.

Reply:
Then let's rename `RESULT_UNDEFINED` to `RESULT_NOT_STARTED`.
Maybe we can also change the naming in all other enums: `UNSPECIFIED` -> `INVALID`

Reply:
proto linter requires 0 enum values to end with `UNSPECIFIED` so we would have to use `RESULT_INVALID_UNSPECIFIED`, wdyt?
 Re: `RESULT_UNDEFINED`, I'm not sure if `RESULT_NOT_STARTED` is the right term, what about `RESULT_UNFINALIZED`?

Reply:
Ah, I didn't know about that linter rule. Then let's keep as is for unspecified.

`RESULT_UNFINALIZED` sounds better :+1: